contracts/cw721-metadata-onchain/src/lib.rs

```diff
@@ -76,22 +94,26 @@ mod tests {
         let mint_msg = MintMsg {
             token_id: token_id.to_string(),
             owner: "john".to_string(),
-            name: "Starship USS Enterprise".to_string(),
-            description: Some("Spaceship with Warp Drive".into()),
-            image: None,
-            extension: Extension {
-                metadata_uri: "http://starships.example.com/Starship/Enterprise.json".into(),
-            },
+            token_uri: Some("https://starships.example.com/Starship/Enterprise.json".into()),
+            extension: Some(Metadata {
+                image: None,
```

Comment:
This is a lot of `None`s... to save your fingers, I would:

1. `#[derive(Default)]` for Metadata and Trait
2. Auto-fill default fields:

```rust
Some(Metadata{
    description: Some("Spaceship with Warp Drive".into()),
    name: Some("Starship USS Enterprise".to_string()),
    ..Metadata::default()
})
```

Reply:
This doesn't change the function at all, but is nice to read and write

Reply:
will do in a later PR..